kuma/users/views.py

```diff
@@ -175,36 +177,23 @@ def profile_edit(request, username):
 
         # Don't validate if the username hasn't changed so people
         # can keep already existing invalid usernames.
-        posted_username = request.POST.get('profile-username', None)
-        if posted_username is not None:
-            username_changed = request.user.username != posted_username
-        else:
-            username_changed = False
+        # posted_username = request.POST.get('user-username', None)
+        # if posted_username is not None:
+        #     username_changed = request.user.username != posted_username
+        # else:
+        #     username_changed = False
 
-        if profile_form.is_valid() and newsletter_form.is_valid():
-            if username_changed:
-                profile.user.username = profile_form.cleaned_data['username']
-                profile.user.save()
+        if user_form.is_valid() and newsletter_form.is_valid():
+            # if username_changed:
+            #     edit_user.username = user_form.cleaned_data['username']
+            #     edit_user.save()
```

Comment:
I checked and there are still >2800 users with legacy usernames such as email addresses or full names with spaces. Unless we want to port those over to slugified usernames (and break a ton URLs) I'm afraid we need to allow having them, but we can enforce the new username policy when changed.